lib/mayaUsd/base/id.h

```diff
@@ -0,0 +1,145 @@
+//
+// Copyright 2020 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#ifndef MAYAUSD_ID_H
+#define MAYAUSD_ID_H
+
+#include <climits>
+#include <string>
+
+namespace MAYAUSD_NS_DEF {
+
+/*! \class Id
```

Comment:
We have been starting to use "version" numbers in base proxy shape and VP2RenderDelegate. Tracking what different int's and long's mean can be tricky, so we are bringing this little class to solve it.